src/components/modules/Clients/NewClientFormModal.tsx

```diff
@@ -0,0 +1,255 @@
+import { TextField } from '@mui/material';
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import Typography from '@mui/material/Typography';
+import { useContext, useEffect, useState } from 'react';
+import colors from '../../../colors';
+import { SnackbarContext } from '../../../hooks/snackbarContext';
+import useHttp from '../../../hooks/useHttp';
+import { CompanyEntity } from '../../../types/company';
+import { RequestMethods } from '../../../utils/constants';
+import CancelButton from '../../common/CancelButton';
+import CreateClientButton from './CreateClientButton';
+
+/**
+ * @brief style object for box component
+ */
+const style = {
+  position: 'absolute' as const,
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 620,
+  bgcolor: 'background.paper',
+  border: '2px solid' && colors.darkGold,
+  boxShadow: 24,
+  p: 4,
+  borderRadius: 3,
+};
+
+interface NewClientFormModalProps {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  setRefetch: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const NewClientFormModal = ({ open, setOpen, setRefetch }: NewClientFormModalProps) => {
+  const [companyName, setCompanyName] = useState('');
+  const [companyEmail, setCompanyEmail] = useState('');
+  const [companyPhone, setCompanyPhone] = useState('');
+  const [companyRFC, setCompanyRFC] = useState('');
+  const [companyConstitution, setCompanyConstitution] = useState('');
+  const [companyTaxResidence, setCompanyTaxResidence] = useState('');
+
+  const { setState } = useContext(SnackbarContext);
+
+  /**
+   * @brief using http hook being able to send new client information
+   */
+  const { data, error, loading, sendRequest } = useHttp<CompanyEntity>(
+    '/company/new',
+    RequestMethods.POST
+  );
+
+  useEffect(() => {
+    handleError();
+    handleSuccess();
+  }, [data, error]);
```

Comment:
GH marca este error: 

React Hook useEffect has missing dependencies: 'handleError' and 'handleSuccess'. Either include them or remove the dependency array

Reply:
listo